include/circt/Dialect/FIRRTL/Passes.td

```diff
@@ -850,4 +850,24 @@ def Lint :
   let constructor = "circt::firrtl::createLintingPass()";
 }
 
+def SpecializeOption :
+    Pass<"firrtl-specialize-option", "firrtl::CircuitOp"> {
+  let summary = "Specialize a configurable design for a given option.";
+  let description = [{
+    If the design has option groups, instead of preserving the information
+    up to the emitted SystemVerilog, this pass specializes it early on for a
+    given choice.
+  }];
+  let constructor = "circt::firrtl::createSpecializeOptionPass()";
+
+  let options = [
+    Option<"selectedOption", "option", "std::string", "", "Option to specialize">,
+    Option<"selectedCase", "case", "std::string", "", "Case to pick">,
+  ];
```

Comment:
What about multiple options? E.g., how would one specialize the test to be "Platform=FPGA" _and_ "Performance=Fast"?

Is this multiple options to the pass or is this handled by dynamically scheduling multiple passes to resolve things iteratively?

Reply:
In the current setup I'd schedule the pass multiple times. In a subsequent PR, when needed, I'd consider compacting the arguments and turning them into an array, but I think that's a bit clumsy with MLIR at the moment.

Reply:
The pass takes a list now.